### ISSUE TITLE
issue regarding the case 2 pretrained model

### ISSUE THREAD

**Issue by MLbeginner3788** [jx-wang-s-group/CoNFiLD#4]
hello,
I was trying to load the diffusion model for case 2, using the ema.pt file and the max and min files which are probably used to scale the data have same value. Also I would like get some help from you to load this model. 
please correct me if am wrong,  
thank you

**Comment by meetcfd** [comment]
Hey, thanks for being interested in our work!

Can you please elaborate on the issue you're facing while using the code?

**Comment by MLbeginner3788** [comment]
Hello sir,
I have downloaded your data for training the Case2 and I could see a file
named output.npy.
When I tried to plot the usual statistics plots like log law and Reynolds
stress,I found the plots to be not similar with what I had expected from
the results of your paper.
I got something like the plots attached below.
So when I generated the data from the trained model, the velocity contours
looked fine but the statistics of the data were not resembling the paper.
Kindly assist me if I am doing something wrong.
the code I used for this is as follows:

import numpy as np
import matplotlib.pyplot as plt

# === Load denormalized data ===
path =
"/workspace/raid/aseris/Harshith/ConFiLD_a/data_confild/data/Case2/output.npy"
data = np.load(path)  # shape: (T, Z, Y, 4)

# === Extract u-velocity
u = data[..., 0]  # (T, Z, Y)

# === Parameters for wall units
Re_tau = 180
nu = 1.0
delta = 1.0
u_tau = Re_tau * nu / delta  # = 180

# === y coordinates in wall units
Y = data.shape[2]
y = np.linspace(0, 1, Y)
y_plus = y * Re_tau

# === Compute mean profile (time + spanwise average)
u_mean = u.mean(axis=(0, 1))  # shape: (Y,)
u_plus = u_mean / u_tau

# === Plot u+ vs y+
plt.figure(figsize=(6, 4))
plt.semilogx(y_plus, u_plus, label=r"$u^+ \text{ vs } y^+$", marker="o",
markersize=3, linewidth=1.5)

# # === Add log-law reference line
# kappa = 0.41
# B = 5.0
# log_y = y_plus[(y_plus > 30) & (y_plus < 200)]
# log_u = (1 / kappa) * np.log(log_y) + B
# plt.semilogx(log_y, log_u, 'k--', label='Log-law: $u^+ =
\\frac{1}{\\kappa} \\ln y^+ + B$')

plt.xlabel(r"$y^+$")
plt.ylabel(r"$u^+$")
plt.title("Mean Velocity Profile")
plt.legend()
plt.grid(True)
plt.tight_layout()
plt.show()

# === Compute Reynolds stresses: u'u', v'v', w'w', u'v'
v = data[..., 1]
w = data[..., 2]

u_fluct = u - u.mean(axis=0, keepdims=True)
v_fluct = v - v.mean(axis=0, keepdims=True)
w_fluct = w - w.mean(axis=0, keepdims=True)

uu = (u_fluct ** 2).mean(axis=(0, 1)) / (u_tau ** 2)
vv = (v_fluct ** 2).mean(axis=(0, 1)) / (u_tau ** 2)
ww = (w_fluct ** 2).mean(axis=(0, 1)) / (u_tau ** 2)
#uv = (u_fluct * v_fluct).mean(axis=(0, 1)) / (u_tau ** 2)

# === Plot Reynolds stress components
plt.figure(figsize=(6, 4))
plt.plot(uu, y_plus, label=r"$\overline{u'^2}^+$")
plt.plot(vv, y_plus, label=r"$\overline{v'^2}^+$")
plt.plot(ww, y_plus, label=r"$\overline{w'^2}^+$")
#plt.plot(-uv, y_plus, label=r"$-\overline{u'v'}^+$")  # usual sign
convention

plt.xlabel(r"Reynolds stress components (normalized)")
plt.ylabel(r"$y^+$")
plt.title("Reynolds Stress Profiles")
plt.legend()
plt.grid(True)
plt.tight_layout()
plt.show()

**Comment by Xin-yang-Liu** [comment]
Thanks for your interest! Please note our saved data `output.npy` for case 2 is of shape [t, y, z, 4]. Simply mean over axis `(0, 2)` should solve your problem.
